### ISSUE TITLE
[Accessibility issue] Dropdown keyboard accessibility regression with up / down keys

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#84652]
**Steps to reproduce**:
Go to any instance of Grafana (such as play.grafana.com) and tab to a dropdown menu and hit space /  enter. Use the up / down arrows to navigate the options. Notice nothing happens.

**Actual Result**
Nothing happens (sometimes it works correctly the first time on a fresh window but it consistently never works in any subsequent tries).

**Expected Result**
I would expect focus to move to the first menu item automatically on toggle of the dropdown opening and the up / down arrows to work correctly.

Currently you have to tab to the menu itself (which has a tabIndex of -1 and it has no focus indicator...) before you can begin using the up / down keys. If you use a mouse to click the toggle then the up / down keys work immediately.

On that note, if you have focus on the dropdown toggle and then hover it with your mouse it loses its focus styles.

It also means that after opening the menu you have to tab multiple times to get to the next element as you tab back to the dropdown toggle, rather than skipping to the next element.

Notice in this video of the behaviour it works as expected the first time round  but not subsequent times.

https://github.com/grafana/grafana/assets/6906380/3ab9ae06-96b9-4a79-b694-303b9493efb7
